feat: add STS for agones (#8)
GameFabric/docs: src/multiplayer-servers/multiplayer-services/server-protection/protocols.md

Some Mitigations only support specific network protocols, such as UDP and TCP.
In those cases, it is impossible to select a network protocol that is not supported by the Mitigation you selected.

![create-protocol.png](images/create-protocol.png)

::: warning
You can restrict your Protocol to only support either UDP or TCP, even though the Mitigation would allow both.
Allowing both protocols while only one is supported by the Mitigation is not allowed.
:::

Once the Protocol is created, you may use it in any ArmadaSet, Armada, Formation or Vessel.
In order to do so, in their Container configuration page, you must select a Protection Protocol for each game server port.

## Protect Game Server Ports

Any ArmadaSet, Armada, Formation or Vessel requires at least one game server port. 
This port is usually meant for communication between your game server and game clients.
In most cases however, your game server exposes more than a single port.
Each of those ports can be configured to use a different Protocol.

In order to make all of your game server's ports secure, it is strongly recommended to apply a Protection Protocol to each of them.

> When the port is unprotected (`none`), traffic is forwarded as-is.

Reviewer: It should be specified that leaving it unprotected would leave the vulnerable to DDoS attacks, negating any other DDoS mitigations that may have been specified.

> When the port is unprotected ('none'), traffic is forwarded as-is with no DDoS protection in place, leaving your server vulnerable to attacks despite protection applies to the other ports.

In most cases however, your game server exposes more than a single port.
Each of those ports can be configured to use a different Protocol.

In order to make all of your game server's ports secure, it is strongly recommended to apply a Protection Protocol to each of them.

When the port is unprotected (`none`), traffic is forwarded as-is with no DDoS protection in place, 
leaving your server vulnerable to attacks despite protection applies to the other ports.

You can find the Protocol assignment to the right of the port specification for your containers.

![game-server-ports.png](images/game-server-ports.png)

::: warning
The assignment triggers an immediate rollout with the new port assignment.
Allocated game servers as always are not affected, so they keep running under the old setup.
:::

## Unsupported Sites

Sites are game server clusters, and SteelShield is not supported on all Sites. 
If you scale up and end up having game servers on multiple Sites, you may have game servers that are protected,
and game servers that are not protected, even if they run under the same revision.

We are working on a feature to provide insights on whether an ArmadaSets,

> Armada, Formation or Vessel is protected, partially protected, or unprotected.

Reviewer 2: I'd remove these two lines, they might be confusing and this feature is the next one in the pipe, so this will be outdated very quickly.

# Ports and Protocols

Game server ports are often vulnerable to attacks. 

> GameFabric offers so-called Protocols to protect your ports with powerful SteelShield mitigations.

Reviewer 3: `so-called` seems to work against us, as it is largely used sarcastically in every day speech.
```suggestion
GameFabric offers Protocols to protect your game server traffic with powerful SteelShield mitigations.
```

## Pre-requisites

In order to protect the your game server traffic, your GameFabric installation requires:

- Protection support enabled, usually indicated by the presence of the Protection/SteelShield navigation item.

> - Sites that support SteelShield, e.g. bare-metal.

Author: I didn't want to be too specific, but give an idea why some sites support it, while others don't. It might also change and gets outdated pretty quick, at least I assume so. And I also don't want to give attackers an indicator which game servers would be more vulnerable as this documentation is also public. 

WDYT?

> - Sites that support SteelShield, e.g. bare-metal.

Reviewer 4: Then I would suggest just removing the `e.g. bare-metal`

## Mitigations

> SteelShield provides mitigations that are tailored to check and let through only specific game or network traffic.

Reviewer 4: > game or network traffic

Weird to me since game traffic is network traffic.

```suggestion
SteelShield provides mitigations that are tailored to check and only allow specific network traffic.
```

Or if we want to make it clear that it could be for specific game traffic:

```suggestion
SteelShield provides mitigations that are tailored to check and only allow specific traffic, such as your game server's protocols.
```

To make use of the pre-defined mitigations you must create a new Protocol.

## Protocols

> A Protocol – sometimes referred to as Protection Protocol, to distinguish it from a 

Author: I used it first, and it was so long, guess that's its purpose, but still :D Anyway trust you in that more than me.

network protocol such as UDP or TCP –

> describes the application protocol used by your game traffic or service.

Reviewer 4: ```suggestion
describes the application protocol used by your game or service.
```

Not sure if you meant to have traffic here?

When you create a new Protocol, you must select a Mitigation and a network protocol.

> The network protocol that you select must be compatible with the Mitigation and their network protocol in order to be selectable.

Reviewer 4: "The network protocol must be compatible with another network protocol" That is confusing, it is explained 2 paragraphs later but I would suggest changing it here:

```
Some Mitigations only support specific network protocols, such as UDP and TCP.
In those cases, it is impossible to select a network protocol that is not supported by the Mitigation you selected.
```

WDYT? IMO clears up the topic and even makes the other paragraphs redundant.

Author: Sounds good, applied.

![create-protocol.png](images/create-protocol.png)

::: warning
You can restrict your Protocol to only support one UDP or TCP, even though the Mitigation would allow both.

> The other way around – allow both where only one is supported by the Mitigation – is not allowed.

Reviewer 4: ```suggestion
Allowing both protocols while only one is supported by the Mitigation is not allowed.
```

:::

> Once the Protocol is created, it is available to all ArmadaSets, Armadas, Formations and Vessels.
> Nonetheless you need to actively select a Protection Protocol for each game server port of these entities.

Reviewer 4: ```suggestion
Once the Protocol is created, you may use it in any ArmadaSet, Armada, Formation or Vessel.
In order to do so, in their Container configuration page, you must select a Protection Protocol for each game server port.
```

## Protect game server ports

Any ArmadaSet, Armada, Formation or Vessel requires at least one game server port. 
This port is usually to allow communication between your game server and your connecting game clients.
It is however common to use multiple ports, one for the game engine, one for query services and so on.
For each of them you can apply different Protocols.

> All ports are potentially vulnerable to attack, so by setting a Protection Protocol,
> SteelShield applies a mitigation action to the traffic on these ports.

Reviewer 4: ```suggestion
In order to make all of your game server's ports secure, it is strongly recommended to apply a Protection Protocol to each of them.
```

> When the port is unprotected (`none`), SteelShield forwards any traffic without checking or filtering it.

Reviewer 4: ```suggestion
When the port is unprotected (`none`), traffic is forwarded as-is.
```

You can find the Protocol assignment to the right of the port specification under 

> Vessel » Settings » Containers » Container Ports. It is the same for ArmadaSets, Armadas and Formations.

Reviewer 4: This isn't specific to Vessels, and I would suggest not including a path here, since it is likely that section names and structure will change and we will not think to update the docs (for reference, see Keycloak documentation, all mentions of where to do things are wrong and they make me want to throw my computer out of the window)

Author: Yeah I am in a struggle between showing the reader where he can actually configure it, and not being required to update the docs all the time to reflect the current GF nav structure. 

Anyway cropped off the sentence a bit. WDYT?

![game-server-ports.png](images/game-server-ports.png)

> You are only allowed to select Protection Protocols that are compatible with the selected network protocol (UDP, TCP).

Reviewer 4: Not sure if this is relevant, seems obvious to me, if a customer sees they can't select their UDP protection protocol for their TCP port and they can't figure it out, I don't think they'll be smart enough to read the docs :p

::: warning

> The assignment will trigger a rollout with the new port assignment.
> Allocated game servers as always are not affected, so they keep running under the old setup.

Reviewer 4: ```suggestion
The assignment triggers an immediate rollout with the new port assignment.
Allocated game servers as always are not affected, so they keep running under the old setup.
```
